test/parallel/test-timers-api-refs.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+// don't verify the globals for this test
+common.globalCheck = false;
+
+// try overriding global APIs to make sure
+// they're not relied on by the timers
+global.clearTimeout = assert.fail;
```

Comment:
I think we should probably add everything here: https://github.com/nodejs/node/blob/ec6e5c79993599a8b6977050bcc09b32b187a8ac/src/node.js#L171-L176